### ISSUE TITLE
Re-organize documentation

### ISSUE THREAD

**Issue by niknetniko** [dodona-edu/dodona-edu.github.io#247]
As discussed, we are looking into merging the TESTed documentation together with the other documentation (e.g. TESTed guides go under Guides, TESTed references go under references, etc.).

I this issue I want to propose (and discuss) a "small" re-organisation of the documentation.

One thing that I noticed is that the sidebar is relatively long, especially if we add TESTed things (and more other doucmentation).
A possible solution would be to no longer duplicate the Guides/References in the sidebar, but only show the Guides when using the Guides and only showing the References when using the references (similar to how https://vitepress.dev/ works).

For example the Guides:
![image](https://github.com/dodona-edu/dodona-edu.github.io/assets/1756811/894cd2e2-5260-431e-b3ee-c91c72560990)

Below are some examples of how the full sidebar might look when applying this. Of course, the order or exact titles of pages is just to demonstrate.

For Guides:

![image](https://github.com/dodona-edu/dodona-edu.github.io/assets/1756811/743f300b-9c70-4cf5-9f79-036a0eea818f)

For References:
![image](https://github.com/dodona-edu/dodona-edu.github.io/assets/1756811/cb618b0f-c388-4023-8e74-1745f3bb955a)

For _reference_ this is the current sidebar:
![image](https://github.com/dodona-edu/dodona-edu.github.io/assets/1756811/26b7d92a-5fa5-4934-a686-e4db0aec662c)

**Comment by bmesuere** [comment]
I agree with the proposed changes. In a next phase, we could re-evaluate the existing pages. For example, "de programmeursleerling" probably shouldn't have his own page. It would be better to instead have a page on featured courses.

**Comment by pdawyndt** [comment]
LGTM
